### PR TITLE
fix adminhtml file attribute edit form

### LINKED ISSUE #11252: Custom attribute - File not allowing uploads
### Preconditions

1. Magento 2.1.9
2. PHP 7.0.2
3. MySQL 5.7

### Steps to reproduce

1. Create a customer file upload attribute
2. View the customer information in the CMS
3. upload a file - in my case a PDF

### Expected result

1. the uploaded file appears in the customer form

### Actual result

1. Popup appears stating something went wrong with saving the file
2. In the apache error log the following appears

[2017-10-05 14:37:15] main.CRITICAL: Exception: Notice: Undefined index: path in /var/www/vhosts/Rosita/public/vendor/magento/module-customer/Model/FileUploader.php on line 113 in /var/www/vhosts/Rosita/public/vendor/magento/framework/App/ErrorHandler.php:61
Stack trace:
#0 /var/www/vhosts/Rosita/public/vendor/magento/module-customer/Model/FileUploader.php(113): Magento\Framework\App\ErrorHandler->handler(8, 'Undefined index...', '/var/www/vhosts...', 113, Array)
#1 /var/www/vhosts/Rosita/public/vendor/magento/module-customer/Controller/Adminhtml/File/Customer/Upload.php(85): Magento\Customer\Model\FileUploader->upload()
#2 /var/www/vhosts/Rosita/public/vendor/magento/framework/App/Action/Action.php(102): Magento\Customer\Controller\Adminhtml\File\Customer\Upload->execute()
#3 /var/www/vhosts/Rosita/public/vendor/magento/module-backend/App/AbstractAction.php(226): Magento\Framework\App\Action\Action->dispatch(Object(Magento\Framework\App\Request\Http))
#4 /var/www/vhosts/Rosita/public/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Backend\App\AbstractAction->dispatch(Object(Magento\Framework\App\Request\Http))
#5 /var/www/vhosts/Rosita/public/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Customer\Controller\Adminhtml\File\Customer\Upload\Interceptor->___callParent('dispatch', Array)
#6 /var/www/vhosts/Rosita/public/vendor/magento/framework/Interception/Chain/Chain.php(63): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Custome...', 'dispatch', Object(Magento\Customer\Controller\Adminhtml\File\Customer\Upload\Interceptor), Array, 'adminAuthentica...')
#7 /var/www/vhosts/Rosita/public/vendor/magento/module-backend/App/Action/Plugin/Authentication.php(143): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#8 /var/www/vhosts/Rosita/public/vendor/magento/framework/Interception/Chain/Chain.php(67): Magento\Backend\App\Action\Plugin\Authentication->aroundDispatch(Object(Magento\Customer\Controller\Adminhtml\File\Customer\Upload\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#9 /var/www/vhosts/Rosita/public/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Custome...', 'dispatch', Object(Magento\Customer\Controller\Adminhtml\File\Customer\Upload\Interceptor), Array, 'adminMassaction...')
#10 /var/www/vhosts/Rosita/public/vendor/magento/module-backend/App/Action/Plugin/MassactionKey.php(33): Magento\Customer\Controller\Adminhtml\File\Customer\Upload\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#11 /var/www/vhosts/Rosita/public/vendor/magento/framework/Interception/Interceptor.php(142): Magento\Backend\App\Action\Plugin\MassactionKey->aroundDispatch(Object(Magento\Customer\Controller\Adminhtml\File\Customer\Upload\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#12 /var/www/vhosts/Rosita/public/var/generation/Magento/Customer/Controller/Adminhtml/File/Customer/Upload/Interceptor.php(26): Magento\Customer\Controller\Adminhtml\File\Customer\Upload\Interceptor->___callPlugins('dispatch', Array, Array)
#13 /var/www/vhosts/Rosita/public/vendor/magento/framework/App/FrontController.php(55): Magento\Customer\Controller\Adminhtml\File\Customer\Upload\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#14 /var/www/vhosts/Rosita/public/vendor/magento/framework/Interception/Interceptor.php(74): Magento\Framework\App\FrontController->dispatch(Object(Magento\Framework\App\Request\Http))
#15 /var/www/vhosts/Rosita/public/vendor/magento/framework/Interception/Chain/Chain.php(70): Magento\Framework\App\FrontController\Interceptor->___callParent('dispatch', Array)
#16 /var/www/vhosts/Rosita/public/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'install')
#17 /var/www/vhosts/Rosita/public/vendor/magento/framework/Module/Plugin/DbStatusValidator.php(69): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Framework\App\Request\Http))
#18 /var/www/vhosts/Rosita/public/vendor/magento/framework/Interception/Interceptor.php(142): Magento\Framework\Module\Plugin\DbStatusValidator->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#19 /var/www/vhosts/Rosita/public/var/generation/Magento/Framework/App/FrontController/Interceptor.php(26): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, Array)
#20 /var/www/vhosts/Rosita/public/vendor/magento/framework/App/Http.php(135): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#21 /var/www/vhosts/Rosita/public/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#22 /var/www/vhosts/Rosita/public/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#23 {main} [] []

In vendor/magento/module-customer/Model/FileProcessor.php line 199 there is unset($result['path']);
This is required by the upload function in vendor/magento/module-customer/Model/FileUploader.php

commenting out line 199 results in the file appearing in the customer form, however clicking the link results in a 404 error

### PR DESCRIPTION
**Pull request by fsw**
fixes #11252

### Description
not sure why this line was introduced as the only place this function is used seems to be here:
https://github.com/magento/magento2/blob/55e9a5cc9d88ef785f39217e96da02b1cdf3e247/app/code/Magento/Customer/Model/FileUploader.php#L110
and it uses 'path' key right after call.
hence no tests are included.

### Fixed Issues (if relevant)
1. magento/magento2#11252: Custom attribute - File not allowing uploads

### Manual testing scenarios
1. Create customer attribute with type file
2. Got to Customer edit form Account Information tab
3. Upload works instead of showing error

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=11267) <br/>All committers have signed the CLA.

File: app/code/Magento/Customer/Model/FileProcessor.php [hunk 1]
```diff
@@ -181,7 +181,6 @@ public function saveTemporaryFile($fileId)
         );
 
         $result = $uploader->save($path);
-        unset($result['path']);
```
Comment (orlangur) [inline app/code/Magento/Customer/Model/FileProcessor.php]: It was added for security reasons: https://github.com/magento/magento2/blame/2.2-develop/app/code/Magento/Customer/Model/FileProcessor.php#L184

Please figure out another way to fix bug you faced with.

**Review by orlangur — CHANGES_REQUESTED** [review]

**Comment by fsw** [comment]
Hi @orlangur,
Thanks for your reply.

Maybe I am missing something but only place in CE and EE I am seeing that is using this function is this place:

https://github.com/magento/magento2/blob/55e9a5cc9d88ef785f39217e96da02b1cdf3e247/app/code/Magento/Customer/Model/FileUploader.php#L110

And it is using 'path' right after and If I see correctly it is using it to create file path to validate it later. So it needs this data.

Path can be unset after that (in FileUploader::upload()) but without knowing what the security issue was I can only guess, Is MAGETWO-70580 an internal ticket number? Can we know exactly what issue this change fixed?
Otherwise I have no other ideas.

cheers.

**Comment by orlangur** [comment]
So, basically, initial security fix needs to be revised which was reverted by changes in this PR and the correct way of fix needs to be understood.

Unassigning from myself for now, will check it again when I have some spare time.

**Comment by omiroshnichenko** [comment]
Hi, @fsw. After debugging I found that `path` used only [here](https://github.com/magento/magento2/blob/55e9a5cc9d88ef785f39217e96da02b1cdf3e247/app/code/Magento/Customer/Model/FileUploader.php#L112-L113). In comment says that `tmp_name` required for attribute validation, but validation performs before this assignment. So, after I removed usage of `$result['path']`, in line that I mentioned before, it works correct. Could you please revert your changes and remove usage of `$result['path']`?

**Comment by fsw** [comment]
Hi @omiroshnichenko, by "remove usage of $result['path']" do you mean to remove pointed line at all or to change it to something like this:

        $result['tmp_name'] = ltrim($result['file'], '/');

If I recall correctly when I did this and uploaded file via admin form it indeed broke saying tmp_name is undefined or something like that but I can't recall exactly. So I think it is indeed used for some validation. 

Anyway I added you as collaborator so please feel free to do whatever you like with this PR or create a new one. 

cheers.

**Review by omiroshnichenko — APPROVED** [review]

**Pull request merged**